util/feed.go

```diff
@@ -251,122 +114,51 @@ func (f *Feed) Publish(event interface{}) {
 //
 // Events are read from the Subscription's Events channel. Subscribers cannot
 // block each other from receiving events, but should still attempt to consume
-// events in a timely fashion; the Feed will not release the memory for an
-// individual event until every Subscriber has seen it.
-//
-// Attempting to subscribe to a closed Feed will panic.
+// events in a timely fashion; if a Subscriber's (very large) Events channel
+// fills up, a panic may result.
 func (f *Feed) Subscribe() *Subscription {
-	// events and closer are used to manage the interaction between a
-	// Subscription and its backing publisher.
-	events := make(chan interface{})
-	closer := make(chan struct{})
 	sub := &Subscription{
-		Events: events,
-		closer: closer,
-	}
-	p := &publisher{
-		events: events,
-		closer: closer,
 		feed:   f,
+		events: make(chan interface{}, subscriptionChannelSize),
 	}
-	select {
-	case f.subscribe <- p:
-		// Subscription was successfully logged.
-	case <-f.closed:
-		panic("attempt to publish value to a closed Feed")
+
+	f.Lock()
+	if !f.closed {
+		f.subscribers = append(f.subscribers, sub)
+	} else {
+		close(sub.events)
 	}
+	f.Unlock()
```

Comment:
Unless there's a specific reason, why not follow the pattern of using defer to unlock mutexes?